tests/tests/test_mtls.py

```diff
@@ -0,0 +1,323 @@
+            config = json.loads(output)
+            # replace mender.conf with an mTLS enabled one
+            config["ServerURL"] = "https://mtls-ambassador:8080"
+            config["ServerCertificate"] = "/etc/mender/server.crt"
+            config["SkipVerify"] = True
+            config["HttpsClient"] = {
+                "Certificate": "/etc/mender/client.1.%s.crt" % algorithm,
+                "Key": "/etc/mender/client.1.%s.key" % algorithm,
+            }
+            del config["ArtifactVerifyKey"]
+            device.run("rm /data/mender/mender.conf")
+            mender_conf = os.path.join(tmpdir, "mender.conf")
+            with open(mender_conf, "w") as fd:
+                json.dump(config, fd)
+            device.put(
+                os.path.basename(mender_conf),
+                local_path=os.path.dirname(mender_conf),
+                remote_path="/etc/mender",
+            )
+        finally:
+            shutil.rmtree(tmpdir)
+
+        # upload the certificates
+        device.put(
+            "tenant.ca.crt",
+            local_path=os.path.join(
+                os.path.dirname(__file__),
+                "..",
+                "..",
+                "extra",
+                "mtls",
+                "certs",
+                "tenant-ca",
+            ),
+            remote_path="/etc/ssl/certs",
+        )
+        device.put(
+            "server.crt",
+            local_path=os.path.join(
+                os.path.dirname(__file__),
+                "..",
+                "..",
+                "extra",
+                "mtls",
+                "certs",
+                "server",
+            ),
+            remote_path="/etc/mender",
+        )
+        device.put(
+            "cert.crt",
+            local_path=os.path.join(
+                os.path.dirname(__file__), "..", "..", "certs", "storage-proxy",
+            ),
+            remote_path="/etc/mender",
+        )
+        device.run(
+            "cat /etc/mender/server.crt /etc/mender/cert.crt > /tmp/server.crt && mv /tmp/server.crt /etc/mender/server.crt && rm /etc/mender/cert.crt"
+        )
+        device.put(
+            "client.1.%s.crt" % algorithm,
+            local_path=os.path.join(
+                os.path.dirname(__file__),
+                "..",
+                "..",
+                "extra",
+                "mtls",
+                "certs",
+                "client",
+            ),
+            remote_path="/etc/mender",
+        )
+        device.put(
+            "client.1.%s.key" % algorithm,
+            local_path=os.path.join(
+                os.path.dirname(__file__),
+                "..",
+                "..",
+                "extra",
+                "mtls",
+                "certs",
+                "client",
+            ),
+            remote_path="/etc/mender",
+        )
+        device.run("update-ca-certificates")
+
+        # TODO: this is needed to circumnavigate a bug in mender-client, which
```

Comment:
Comment:
This can safely be removed now that we're on the newest sha :)

Reply:
Removed